GVRf/Framework/src/org/gearvrf/GVRMaterial.java

```diff
@@ -267,6 +269,7 @@ public GVRTexture getTexture(String key) {
     }
 
     public void setTexture(String key, GVRTexture texture) {
+        checkKeyNotNullOrEmpty(key);
         NativeMaterial.setTexture(getPtr(), key, texture.getPtr());
```

Comment:
Yes. I think I made a mistake. I thought the crash and error will occurs only when getTexture() of null texture is called. I will update it.